src/hardware/voodoo.cpp

```diff
@@ -53,40 +55,42 @@ enum class CardType : int {
     Type1 = 1
 };
 
-constexpr const char* NONE_STR = "false";
-constexpr const char* SOFTWARE_STR = "software";
-constexpr const char* OPENGL_STR = "opengl";
-constexpr const char* AUTO_STR = "auto";
+constexpr const char* NoneStr = "false";
+constexpr const char* SoftwareStr = "software";
+constexpr const char* OpenGLStr = "opengl";
+constexpr const char* AutoStr = "auto";
 
-class VOODOO : public Module_base {
+class VOODOO {
 private:
+    Section* m_configuration;
     EmulationType emulation_type;
 
 public:
-    VOODOO(Section* configuration) : Module_base(configuration), emulation_type(EmulationType::None) {
+	VOODOO(const VOODOO&) = delete;
+	VOODOO& operator=(const VOODOO&) = delete;
+	
+    VOODOO(Section* configuration) : m_configuration(configuration), emulation_type(EmulationType::None) {
         UpdateConfiguration(configuration);
     }
 	
-	EmulationType GetEmulationTypeFromString(const std::string& voodoo_type_str) {
-		static const std::unordered_map<std::string, EmulationType> str_to_emulation_type {
-			{NONE_STR, EmulationType::None},
-			{SOFTWARE_STR, EmulationType::Software},
-			{OPENGL_STR, C_OPENGL ? EmulationType::OpenGL : EmulationType::None},
-			{AUTO_STR, C_OPENGL ? EmulationType::OpenGL : EmulationType::Software}
-		};
-
-		auto it = str_to_emulation_type.find(voodoo_type_str);
-		if (it != str_to_emulation_type.end()) {
-			return it->second;
+	EmulationType GetEmulationTypeFromString(const char* voodoo_type_str) {
+		if (strcmp(voodoo_type_str, NoneStr) == 0) {
+			return EmulationType::None;
+		} else if (strcmp(voodoo_type_str, SoftwareStr) == 0) {
+			return EmulationType::Software;
+		} else if (strcmp(voodoo_type_str, OpenGLStr) == 0) {
+			return C_OPENGL ? EmulationType::OpenGL : EmulationType::None;
+		} else if (strcmp(voodoo_type_str, AutoStr) == 0) {
+			return C_OPENGL ? EmulationType::OpenGL : EmulationType::Software;
+		} else {
+			return !OpenGL_using() ? EmulationType::Software : EmulationType::None;
```

Comment:
Please use modern C++ strings or string_view so comparisons can be done naturally using `==`.  

https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#contributing-code